src/app/form-register/form-register.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-form-register',
+  templateUrl: './form-register.component.html',
+  styleUrls: ['./form-register.component.scss']
+})
+export class FormRegisterComponent implements OnInit {
+
+  registerForm: FormGroup;
+  post: any;
+  firstName: string = '';
+  lastName: string = '';
+  email: string = '';
+  passwordConfirmation: string = '';
+  messageAlert: string = 'Campo obligatorio'
+
+  constructor( private fb: FormBuilder ) { }
+
+  private buildForm(){
+    this.registerForm = this.fb.group({
+      'firstName': [null, Validators.compose([Validators.required, Validators.minLength(3)])],
+      'lastName': [null, Validators.compose([Validators.required, Validators.minLength(3)])],
+      'email': [null, Validators.compose([Validators.required, Validators.email])],
+      'passwordConfirmation': [null, Validators.compose([Validators.required, Validators.maxLength(20), Validators.minLength(5)])]
+    });
+  }
+
+  addPost(post) {
+    this.firstName = post.firstName;
+    this.lastName = post.lastName;
+    this.email = post.email;
+    this.passwordConfirmation = post.passwordConfirmation;
+    console.log(post);
+  }
+
+  ngOnInit() {
```

Comment:
I always try to put ngOnInit after the constructor.

Reply:
this is a  lifecycle ? @fmontoy